src/test/FuzzerImpl.cpp

```diff
@@ -729,17 +730,76 @@ TransactionFuzzer::initialize()
         genOffersForPair(ASSET_B, ASSET_C, {13, 2, 14, 3, 15}, ltx);
         genOffersForPair(ASSET_C, ASSET_D, {6, 3, 7, 4, 8}, ltx);
 
-        // construct transaction
-        auto tx = createFuzzTransactionFrame(mSourceAccountID, ops,
-                                             mApp->getNetworkID());
+        attemptToApplyOps(ltx, mSourceAccountID, ops, *mApp);
 
-        tx->attemptApplication(*mApp, ltx);
+        ltx.commit();
+    }
 
-        if (tx->getResultCode() != txSUCCESS)
+    {
+        LedgerTxn ltx(ltxOuter);
+
+        xdr::xvector<Operation> ops;
+
+        // Reduce account balances so that fuzzing has a better change
+        // of exercising limits.
+        for (uint8_t i = 1; i < FuzzUtils::NUMBER_OF_PREGENERATED_ACCOUNTS; ++i)
         {
-            throw std::runtime_error("Error while initializing ledger state");
+            PublicKey account;
+            FuzzUtils::setShortKey(account, i);
+
+            // Reduce "account"'s native balance by paying the root.
+            auto ae = stellar::loadAccount(ltx, account);
+            auto const availableBalance =
+                getAvailableBalance(ltx.loadHeader(), ae);
+            auto const targetAvailableBalance =
+                FuzzUtils::DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY +
+                FuzzUtils::FUZZING_FEE *
+                    FuzzUtils::DEFAULT_NUM_TRANSACTIONS_TO_RESERVE_FEES_FOR;
+
+            if (availableBalance > targetAvailableBalance)
+            {
+                auto reduceNativeBalanceOp =
+                    txtest::payment(mSourceAccountID, txtest::makeNativeAsset(),
+                                    availableBalance - targetAvailableBalance);
+                reduceNativeBalanceOp.sourceAccount.activate() =
+                    toMuxedAccount(account);
+                ops.emplace_back(reduceNativeBalanceOp);
+            }
+
+            for (int j = 1; j <= FuzzUtils::NUMBER_OF_ASSETS_TO_ISSUE; ++j)
+            {
+                auto const asset = FuzzUtils::makeAsset(j);
+                if ((i != 0) && (i != j))
```

Comment:
`i == 0` is impossible here. Also you're doing a signed/unsigned comparison `i != j` here which is generally unsafe.

Reply:
Fixed.